src/GafferScene/Instancer.cpp

```diff
@@ -2896,14 +3020,14 @@ void Instancer::InstancerCapsule::render( IECoreScenePreview::Renderer *renderer
 
 				if( sampleTimes.size() == 1 )
 				{
-					objectInterface->transform( proto->m_transforms[0] * engines[0]->instanceTransform( pointIndex ) );
+					objectInterface->transform( proto->m_transforms[0] * engines[0]->instanceTransform( i ) );
 				}
 				else
 				{
-					for( unsigned int i = 0; i < engines.size(); i++ )
+					for( unsigned int j = 0; j < engines.size(); j++ )
 					{
-						int curPointIndex = i == 0 ? pointIndex : engines[i]->pointIndex( instanceId );
-						pointTransforms[i] = proto->m_transforms[i] * engines[i]->instanceTransform( curPointIndex );
+						int curPointIndex = j == 0 ? i : engines[j]->pointIndex( instanceId );
+						pointTransforms[j] = proto->m_transforms[j] * engines[j]->instanceTransform( curPointIndex );
```

Comment:
If we called the previous variable `pointIndex` rather than `i`, could this all stay unchanged?